pennylane_lightning/lightning_kokkos/lightning_kokkos.py

```diff
@@ -294,6 +306,104 @@ def simulate_and_vjp(
 # The set of supported observables.
 
 
+def stopping_condition(op: Operator) -> bool:
+    """A function that determines whether or not an operation is supported by ``lightning.kokkos``."""
+    # These thresholds are adapted from `lightning_base.py`
+    # To avoid building matrices beyond the given thresholds.
+    # This should reduce runtime overheads for larger systems.
+    if isinstance(op, qml.QFT):
+        return len(op.wires) < 10
+    if isinstance(op, qml.GroverOperator):
+        return len(op.wires) < 13
+
+    # As ControlledQubitUnitary == C(QubitUnitrary),
+    # it can be removed from `_operations` to keep
+    # consistency with `lightning_qubit.toml`
+    if isinstance(op, qml.ControlledQubitUnitary):
+        return True
```

Comment:
This is not supported (natively) by L-Kokkos, and hence can be removed.
```suggestion
```

Reply:
Could be better to return `false`? Additionally, LK does not have native support for Controlled gates but it can perform these ops in some way. Therefore why not keep these lines? :thinking:

Reply:
Setting it to `True`, it won't attempt decomposition, which means it will be applied as a matrix. If you have a lot of controls, this is a problem. For instance in a 20-qubit Grover step, you'll need a 20-qubit unitary matrix which will kill the computer's memory. Setting it to `False` is equivalent to `True` it since it will decompose until everything is in `_operations` and `ControlledQubitUnitary` is listed in `_operations`. We should decompose operations with arbitrary controls (or at least if there are more than 3-4 controls) to avoid a performance/memory wall.

Reply:
Done, thank you for the explanation. :smile: